tinygrad/codegen/linearizer.py

```diff
@@ -412,7 +412,11 @@ def uop(self, uop:UOps, dtype:Optional[DType], vin:Tuple[UOp, ...], arg:Any=None
   def ast_parse(self, x, acc, offs, loaded_buffers, do_reduce=False) -> List[UOp]:
     if x.__class__ is not LazyOp: return loaded_buffers[x]    # for LOCAL_BUFFER
     if x.op in BufferOps: return loaded_buffers[x.arg]
-    if x.op in [UnaryOps.NOOP, UnaryOps.CAST]: return self.ast_parse(x.src[0], acc, offs, loaded_buffers)  # cast isn't an ALU op
+    if x.op == UnaryOps.NOOP: return self.ast_parse(x.src[0], acc, offs, loaded_buffers)
+    if x.op == UnaryOps.CAST:
+      dtype, bitcast = x.arg
+      if not bitcast: return self.ast_parse(x.src[0], acc, offs, loaded_buffers)
+      return [self.uop(val.uop, dtype, val.vin, (val.arg, bitcast)) for val in self.ast_parse(x.src[0], acc, offs, loaded_buffers)]
```

Comment:
I don't think this is right. Is there an implicit assumption here that this is a load?

Reply:
yeah there was - Limited it to just LOAD and GEP